main.py

```
@@ -210,6 +221,14 @@ async def handler(data):
 
             logger.info("🔄 Handshake successful with %s", peer_addr)
             peer_tip = payload.get("latest_block_index", 0)
+
+            if not isinstance(peer_tip, int) or isinstance(peer_tip, bool):
+                logger.warning(
+                    "Malformed hello from %s: latest_block_index is not an integer (got %s). Disconnecting.",
+                    peer_addr, type(peer_tip).__name__
+                )
+                asyncio.create_task(network.disconnect_peer(peer_addr))
```

Ruff (0.16.1)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)